packages/salesforcedx-vscode-core/src/commands/forceSourceDeploy.ts

```diff
@@ -24,19 +27,41 @@ import { telemetryService } from '../telemetry';
 import { getRootWorkspacePath } from '../util';
 import { SfdxCommandletExecutor } from './commands';
 
+export interface DeployParams {
+  sourcePush: boolean;
+  sourcePaths: string;
+}
+
+export class DeployParamsGatherer implements ParametersGatherer<DeployParams> {
```

Comment:
Hey Bryan, as an alternative to creating a new `DeployParamsGatherer` and `SourcePathDeployChecker` in order to plumb the `sourcePush` boolean through the entire command, how would you feel about instead having forceSourceDeploy.ts's `outputResult()` call out to another instance method `getTableType()` to get the table type? You could implement something like `protected getTableType() { return 'deploy' }` in ForceSourceDeployExecutor and override the function in ForceSourcePushExecutor to be something like `protected getTableType() { return 'push' };

Reply:
@allileong I like that idea a lot better!